src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs

```diff
@@ -0,0 +1,553 @@
+﻿// <copyright file="ApplicationScenarios.cs" company="Okta, Inc">
+// Copyright (c) 2014 - present Okta, Inc. All rights reserved.
+// Licensed under the Apache 2.0 license. See the LICENSE file in the project root for full license information.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Xunit;
+using System.Linq;
+
+namespace Okta.Sdk.IntegrationTests
+{
+    [Collection(nameof(ScenariosCollection))]
+    public class ApplicationScenarios : ScenarioGroup
+    {
+        [Fact]
+        public async Task AddBookmarkApp()
+        {
+            var client = GetClient();
+
+            var createdApp = await client.Applications.CreateApplicationAsync(new BookmarkApplication
+            {
+                Name = "bookmark",
+                Label = "Sample Bookmark App",
+                SignOnMode = ApplicationSignOnMode.Bookmark,
+                Settings = new BookmarkApplicationSettings
+                {
+                    App = new BookmarkApplicationSettingsApplication
+                    {
+                        RequestIntegration = false,
+                        Url = "https://example.com/bookmark.htm",
+                    },
+                },
+            });
+
+            try
+            {
+                var retrieved = await client.Applications.GetApplicationAsync(createdApp.Id);
```

Comment:
Discuss adding a helper: `client.Applications.GetApplicationAsync<IApplicationType>(appId)` when the application type is already known (like here). This would remove the need for casts in these tests.

Reply:
In other words, I want to be able to do:

```csharp
var retrieved = await client.Applications.GetApplicationAsync<IBookmarkApplication>(id);
```
and have `retrieved` be typed as `IBookmarkApplication`.